crm_stage_type/tests/test_crm_stage_type.py

```diff
@@ -16,26 +16,54 @@ def setUp(self):
             'groups_id': [(6, 0, [self.ref('sales_team.group_sale_manager')])]
         })
 
-    def test_find_stage(self):
-        lead = self.env['crm.lead'].create({
+        Stage = self.env['crm.stage']
+        # archiving default odoo stages is not possible, so change their lead_type
+        # and use Won stage for both types
+        Stage.browse([1, 2, 3]).write({'lead_type': "opportunity"})
+        # Opportunity: New => Qualified => Proposition => Won [both]
+        # Lead: A => B => Won [both]
+        self.lead_stage_a = Stage.create(
+            {'name': "A", 'lead_type': "lead", 'sequence': 4}
+        )
+        self.lead_stage_b = Stage.create(
+            {'name': "B", 'lead_type': "lead", 'sequence': 5}
+        )
+
+        self.lead = self.env['crm.lead'].with_context({
+            'default_type': 'lead',
+        }).create({
             'type': "lead",
             'name': "Test lead new",
             'partner_id': self.env.ref("base.res_partner_1").id,
             'description': "This is the description of the test new lead.",
-            'team_id': self.env.ref("sales_team.team_sales_department").id
+            'team_id': self.env.ref('sales_team.team_sales_department').id,
+
         })
-        self.assertTrue(lead.stage_id.lead_type, 'both')
-
-        lead.convert_opportunity(self.env.ref("base.res_partner_2").id)
-        self.assertLessEqual(lead.stage_id.sequence, 1,
-                             "Default stage of lead is incorrect!")
-        self.assertTrue(lead.stage_id.lead_type, 'both')
-        lead.action_set_won()
-        self.assertTrue(lead.stage_id.lead_type, 'both')
-        stage_id = lead._stage_find(domain=[('probability', '=', 100.0)])
-        self.assertEqual(stage_id, lead.stage_id,
-                         "Stage of opportunity is incorrect!")
-        self.assertTrue(stage_id.lead_type, 'both')
+
+    def test_read_group_stage_ids(self):
+        # test returned stages for lead
+        groups_lead = self.lead._read_group_stage_ids(
+            self.env['crm.stage'].search([]), None, "sequence"
+        )
+        self.assertEqual(
+            groups_lead.ids,
+            [
+                self.lead_stage_a.id,
+                self.lead_stage_b.id,
+                self.env.ref('crm.stage_lead4').id,
+            ],
+        )
+
+    def test_find_stage(self):
+        self.assertTrue(self.lead.stage_id, self.lead_stage_a)
```

Comment:
this assert seems useless as you create them on setup

Reply:
That's the test to be sure that the lead (type=lead) gets the correct default stage (Stage name A)

Reply:
but then you should use assertEqual, here you are testing only that stage_id is valued